Allow relative CLDR paths in the dojorc

<!--
Thank you for contributing to Dojo 2.

Our issue tracker is for bugs for Dojo 2.

Please make sure you have read our Contributing Guidelines
available at: https://github.com/dojo/meta/blob/master/CONTRIBUTING.md

For general questions and discussion, join us on Gitter.im at: https://gitter.im/dojo/dojo2
-->

**Bug**

The `I18nPlugin` currently does not allow any relative value in the `cldrPaths` rc option. This is probably fine in many applications since CLDR data will likely be supplied by a third-party module. However, in the event that CLDR data is provided by the application itself, it would be best if relative paths were assumed to be relative to `process.cwd()`.

**Code**

```
// .dojorc
{
    "build-app": {
        "cldrPaths": [ "./src/nls/cldr-data.json" ]
    }
}
```

**Expected behavior:**

The JSON file `${process.cwd()}/src/nls/cldr-data.json` is bundled with the application.

**Actual behavior:**

The build throws the error, `Cannot find module './src/nls/cldr-data-.json'`

--- a/README.md
+++ b/README.md
@@ -184,7 +184,7 @@ The plugin accepts an options object with the following properties:
 
 | Property | Type | Optional | Description |
 | -------- | ---- | -------- | ----------- |
-| cldrData | string[] | Yes | An array of paths to CLDR JSON modules that should be included in the build and registered with the i18n ecosystem. |
+| cldrData | string[] | Yes | An array of paths to CLDR JSON modules that should be included in the build and registered with the i18n ecosystem. If a path begins with a ".", then it is treated as relative to the current working directory. Otherwise, it is treated as a valid mid. |
 | defaultLocale | string | No | The default locale. |
 | supportedLocales | string[] | Yes | An array of supported locales beyond the default. |
 | target | string | Yes | The entry point into which the i18n module should be injected. Defaults to `src/main.ts`. |

--- a/src/i18n-plugin/I18nPlugin.ts
+++ b/src/i18n-plugin/I18nPlugin.ts
@@ -5,6 +5,8 @@ import { Compiler, DefinePlugin } from 'webpack';
 import NormalModule = require('webpack/lib/NormalModule');
 import InjectedModuleDependency from './dependencies/InjectedModuleDependency';
 
+const basePath = process.cwd();
+
 export interface I18nPluginOptions {
 	/**
 	 * An optional list of CLDR JSON paths used to inject CLDR data into the application.
@@ -109,6 +111,9 @@ export default class I18nPlugin {
 
 		return this.cldrPaths
 			.map((url) => {
+				if (url.charAt(0) === '.') {
+					url = join(basePath, url);
+				}
 				return locales.map((locale) => url.replace('{locale}', locale));
 			})
 			.reduce((left, right) => left.concat(right), [])

--- a/tests/unit/i18n-plugin/I18nPlugin.ts
+++ b/tests/unit/i18n-plugin/I18nPlugin.ts
@@ -105,9 +105,9 @@ describe('I18nPlugin', () => {
 		};
 		const plugin = new I18nPlugin({
 			defaultLocale: 'en',
-			cldrPaths: ['{locale}/main.json', 'supplemental.json'].map((file) =>
-				join(process.cwd(), 'tests/support/fixtures/cldr/', file)
-			)
+			cldrPaths: ['{locale}/main.json', 'supplemental.json'].map((file) => {
+				return `./tests/support/fixtures/cldr/${file}`;
+			})
 		});
 		applyModule(plugin);
 

